### PR TITLE
Create Closed Loop Control.md

### PR DESCRIPTION
**Pull request by jessegriffey**

### CODE REVIEW

File: Documentation/Signoff/Direction Sensor.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# Direction Sensor Subsystem
+
+## Function of the system
+This subsystem is designed so the car will go in a straight line and not veer off course. An IMU chip will be used to help the system know if it is by sending signals. When it senses its going off course, it tells the front two servo motors to adjust the wheels in a certain direction to keep the car on track. So basically, the subsystem makes sure the car goes where it is pointed and stays on that path. It is almost like having an automatic steering system in it.
+
+## Constraints
+The subsystems constraints are as follows:
+
+1.	The weight distribution of the car will not be perfect. One side will weigh more than the other side and this must be accounted for. A car with more weight on one side will naturally turn that direction. So, the program will have to be written to adjust for this weight.
```
Comment (JesseTNRoberts) [inline Documentation/Signoff/Direction Sensor.md]: Not a constraint.

File: Documentation/Signoff/Direction Sensor.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+2.	When the physical part of the car is designed, a place for the IMU chip, even though it is small, must be planned out so it is not in the way of anything else. It also needs to be known that the wires need a mapped-out route to the motor and Arduino. So basically, the car needs a place for the IMU chip and wires to make sure it doesn’t disrupt other subsystems and the functionality of the car.
```
Comment (JesseTNRoberts) [inline Documentation/Signoff/Direction Sensor.md]: Not a constraint.

**Review by JesseTNRoberts — COMMENTED** [review]
The constraints here are largely irrelevant. Further, the presentation is ambiguous and messy. Some of them aren't even constraints.

Do not submit this design to me for review until it has first been thoroughly reviewed internally by your team.

File: Documentation/Signoff/Direction Sensor.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
+# Closed Loop Control
+
+## Function of the system
+The purpose of this subsystem is to drive and steer the vehicle. The subsystem will keep the car in the desired location while the car is moving on the designated mat. It will determine if the car has veered off course and adjust accordingly.
```
Comment (JesseTNRoberts) [inline Documentation/Signoff/Direction Sensor.md]: What mat?

File: Documentation/Signoff/Direction Sensor.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+## Constraints
+| Description | Constraints/Specification | Source |
+|-------------|---------------------------|--------|
+| Voltage | The system shall operate with the correct voltage. The motor driver shield is rated for 4.5V to 13.5V, the IMU chip is rated for 3V-5V, and the TT DC motors are recommended for 6-8V.  | [1], [3], [8] |
+| Current | The motor driver shield must be capable of supplying 150mA of current per bridge for the four motors, while also supporting the IMU chip's maximum draw of 12.3mA. | [1], [3], [8] |
+| Precision | The system must detect if the car is veering off then immediately correct it and put it back on track. The IMU's sampling rate must be around 100Hz to achieve the speed. | [4] |
```
Comment (JesseTNRoberts) [inline Documentation/Signoff/Direction Sensor.md]: How do you know the needed sampling rate? You did not list a precision requirement.

**Review by JesseTNRoberts — COMMENTED** [review]
This is a control system. So, it is similar to sensing with additional considerations. First, how fast does the sensor need to sample and why? How precise does the sensor need to be and why? Is the overall control system stable? 

Lastly and most important, how much error is acceptable?

File: Documentation/Signoff/Direction Sensor.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+## Function of the system
+The purpose of this subsystem is to drive and steer the vehicle. The subsystem will keep the car in the desired location while the car is moving. It will determine if the car has veered off course and adjust accordingly.
+
+## Constraints
+| Description | Constraints/Specification | Source |
+|-------------|---------------------------|--------|
+| Voltage | The system shall operate with the correct voltage. The motor driver shield is rated for 4.5V-13.5V, the IMU chip is rated for 3V-5V, and the TT DC motors are recommended for 6-8V. | [1], [3], [8] |
+| Current | The motor driver shield must be capable of supplying 150mA of current per bridge for the four motors, while also supporting the IMU chip's maximum draw of 12.3mA.  | [1], [3], [8] |
+| Socioeconomic | The system shall be cost effective relating to the project's budget constraints. | Conceptual Design |
```
Comment (JesseTNRoberts) [inline Documentation/Signoff/Direction Sensor.md]: How does this affect the society and/or economy?

File: Documentation/Signoff/Direction Sensor.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+| Response | The system must detect if the car is veering off then immediately correct it and put it back on track. The IMU’s sampling rate must be around 100Hz to achieve the accepted error of 0.033 meters. | [4], [11] |
```
Comment (JesseTNRoberts) [inline Documentation/Signoff/Direction Sensor.md]: Given precise ranges with reasoning below explaining how you know that the range given is appropriate and sufficient.

File: Documentation/Signoff/Direction Sensor.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
+# Closed Loop Control
+
+## Function of the system
+The purpose of this subsystem is to drive and steer the vehicle. The subsystem will keep the car in the desired location while the car is moving. It will determine if the car has veered off course and adjust accordingly.
+
+## Constraints
```
Comment (JesseTNRoberts) [inline Documentation/Signoff/Direction Sensor.md:6]: Constraint origins need to be specified. Not cited here.

File: Documentation/Signoff/Direction Sensor.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+| Description | Constraints/Specification | Source |
+|-------------|---------------------------|--------|
+| Voltage | The system shall operate with the correct voltage. The motor driver shield is rated for 4.5V-13.5V, the IMU chip is rated for 3V-5V, and the TT DC motors are recommended for 6-8V. | [1], [3], [8] |
+| Current | The motor driver shield must be capable of supplying 150mA of current per bridge for the four motors, while also supporting the IMU chip's maximum draw of 12.3mA.  | [1], [3], [8] |
+| Cost Boundaries | Equipment/components should not surpass $35, with the total capped at $90 to ensure the project's final total falls within the $400 to $900 budget. | Conceptual Design |
+| Response | The system must detect if the car is veering off then immediately correct it and put it back on track. With the IMU’s sampling rate of 100Hz and the motors being 0.1 horsepower, the vehicles speed will be between 1 and 9.28 miles per hour but must achieve the accepted error of 0.033 meters. | [3], [4], [11] |
```
Comment (JesseTNRoberts) [inline Documentation/Signoff/Direction Sensor.md]: I don't understand this. There needs to be a justification below that explains where this comes from.

File: Documentation/Signoff/Direction Sensor.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+
+Analyzing the budget cost boundaries: To ensure budget constraints are followed, individual components shall not exceed $35, and the total shall not exceed $90. The $400-$900 budget must be kept in mind for this design. In addition, keeping costs low has good effects on the economy. It shall be cost-effective while also being high in quality. This system is crucial, so it needs to be reliable. 
+
+Analyzing the response constraint: The vehicle should be able to drive many different conditions and go the correct path. The IMU has a sampling rate of 100Hz meaning it takes samples or measurements 100 times a second [2]. This speed will allow the vehicle to make quick adjustments to stay on course based on the calculations and information below. On the control algorithm side, the speed of the motors is set in the program. The vehicles speed does not need to be too fast, or it will turn and adjust too rapidly. 
+
+The TT motors are 0.1 hp [3]. Taking the cube root of the horsepower and multiplying it by 20 gives the max mile per hour of the motors [10].
```
Comment (JesseTNRoberts) [inline Documentation/Signoff/Direction Sensor.md]: If you know the motor, why are you not using the rated speed of the motor?

File: Documentation/Signoff/Direction Sensor.md [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+
+This is the speed for just the motors, not while they are on the vehicle. When on the vehicle, it will not go this speed due to the circumstances of the vehicles weight, wheel friction, terrain, etc. By doing calculations, we can find out the range of speed needed based on the 100Hz sampling rate and how much error is acceptable. By using 5mph as an example, the first step is finding the distance covered per sample:
+
+$\frac{5*5280}{3600}$ = 7.33ft/s (1 mile = 5280ft) (1 hour = 3600 seconds)
+
+Then, find the distance covered per sample: 
+
+$7.33 \times \frac{1}{100}$ = 0.0733ft/sample (Velocity * Time Interval)
+
+So every 100Hz sample the vehicle will cover 0.0733 feet. An accepted width of error is 33mm or 0.033 meters [11]. By using this, we can calculate if 5 miles per hour works for the width of error:
```
Comment (JesseTNRoberts) [inline Documentation/Signoff/Direction Sensor.md]: What is a width error?

File: Documentation/Signoff/Direction Sensor.md [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+## Bill of Materials (BOM)
+| Item Name | Details | Quantity | Cost | Source |
+|-----------|---------|----------|------|--------|
+| Arduino Uno | Rev 3 | 1 | $27.60 | https://shorturl.at/Oy4Ch |
+| IMU Chip| BNO055| 1 | $29.95 | https://shorturl.at/tnMpC |
+| Arduino Motor Driver Shield | Adafruit| 1 | $19.95 | https://shorturl.at/0jVxw |
+| DC motors/Wheels | TT Gearbox motors | 4 | $8.99 | https://shorturl.at/hUIP2 |
+| Switch | Criotech | 1 | $1.68 | https://tinyurl.com/4v464t8m |
+| Connector | Unpluggable wire connector | 1 | $0.59 | https://tinyurl.com/39tmh9n4 |
+| Jumper Wires | Female/male wires 20 x 3" | 1 | $1.95 | https://tinyurl.com/2zdpf96w |
```
Comment (micahr234) [inline Documentation/Signoff/Direction Sensor.md]: You need to specify the part number and manufacture on your BOM.

**Review by micahr234 — APPROVED** [review]

**Review by micahr234 — APPROVED** [review]

**Pull request merged**